[bug] Investigate occasional panic in tests

```
=== CONT  TestServerSideRefreshCustomError
panic: close of closed channel

goroutine 464241 [running]:
github.com/centrifugal/centrifuge.TestServerSideRefreshCustomError.func3.1({0x40?, {0x0?, 0x4ae2c5?}}, 0xc0083b5e40)
	/home/runner/work/centrifuge/centrifuge/client_test.go:2726 +0x157
github.com/centrifugal/centrifuge.(*Client).expire(0xc00da28640)
	/home/runner/work/centrifuge/centrifuge/client.go:1388 +0x262
github.com/centrifugal/centrifuge.(*Client).onTimerOp(0xc00da28640)
	/home/runner/work/centrifuge/centrifuge/client.go:419 +0x1a5
created by time.goFunc
	/opt/hostedtoolcache/go/1.20.14/x64/src/time/sleep.go:176 +0x48
FAIL	github.com/centrifugal/centrifuge	47.623s
```

--- a/client.go
+++ b/client.go
@@ -496,7 +496,7 @@ func (c *Client) sendPing() {
 	if c.pongTimeout > 0 && !unidirectional {
 		c.nextPong = time.Now().Add(c.pongTimeout).UnixNano()
 	}
-	c.addPingUpdate(false)
+	c.addPingUpdate(false, true)
 	c.mu.Unlock()
 }
 
@@ -519,7 +519,7 @@ func (c *Client) checkPong() {
 }
 
 // Lock must be held outside.
-func (c *Client) addPingUpdate(isFirst bool) {
+func (c *Client) addPingUpdate(isFirst bool, scheduleNext bool) {
 	delay := c.pingInterval
 	if isFirst {
 		// Send first ping in random interval between 0 and PingInterval to
@@ -529,19 +529,25 @@ func (c *Client) addPingUpdate(isFirst bool) {
 		delay = time.Duration(randSource.Int63n(pingNanoseconds)) * time.Nanosecond
 	}
 	c.nextPing = time.Now().Add(delay).UnixNano()
-	c.scheduleNextTimer()
+	if scheduleNext {
+		c.scheduleNextTimer()
+	}
 }
 
 // Lock must be held outside.
-func (c *Client) addPresenceUpdate() {
+func (c *Client) addPresenceUpdate(scheduleNext bool) {
 	c.nextPresence = time.Now().Add(c.node.config.ClientPresenceUpdateInterval).UnixNano()
-	c.scheduleNextTimer()
+	if scheduleNext {
+		c.scheduleNextTimer()
+	}
 }
 
 // Lock must be held outside.
-func (c *Client) addExpireUpdate(after time.Duration) {
+func (c *Client) addExpireUpdate(after time.Duration, scheduleNext bool) {
 	c.nextExpire = time.Now().Add(after).UnixNano()
-	c.scheduleNextTimer()
+	if scheduleNext {
+		c.scheduleNextTimer()
+	}
 }
 
 // closeStale closes connection if it's not authenticated yet, or it's
@@ -703,7 +709,7 @@ func (c *Client) updatePresence() {
 		}
 	}
 	c.mu.Lock()
-	c.addPresenceUpdate()
+	c.addPresenceUpdate(true)
 	c.mu.Unlock()
 }
 
@@ -1333,7 +1339,7 @@ func (c *Client) checkExpired() {
 		if ttl > 0 {
 			c.mu.Lock()
 			if c.status != statusClosed {
-				c.addExpireUpdate(time.Duration(ttl) * time.Second)
+				c.addExpireUpdate(time.Duration(ttl)*time.Second, true)
 			}
 			c.mu.Unlock()
 		}
@@ -1418,18 +1424,20 @@ func (c *Client) triggerConnect() {
 func (c *Client) scheduleOnConnectTimers() {
 	// Make presence and refresh handlers always run after client connect event.
 	c.mu.Lock()
-	c.addPresenceUpdate()
+	c.addPresenceUpdate(false)
 	if c.exp > 0 {
 		expireAfter := time.Duration(c.exp-time.Now().Unix()) * time.Second
 		if c.clientSideRefresh {
 			conf := c.node.config
 			expireAfter += conf.ClientExpiredCloseDelay
 		}
-		c.addExpireUpdate(expireAfter)
+		c.addExpireUpdate(expireAfter, false)
 	}
 	if c.pingInterval > 0 {
-		c.addPingUpdate(true)
+		c.addPingUpdate(true, false)
 	}
+	// Only schedule next timer once here after setting required points in time for ops.
+	c.scheduleNextTimer()
 	c.mu.Unlock()
 }
 
@@ -1466,7 +1474,7 @@ func (c *Client) Refresh(opts ...RefreshOption) error {
 				c.info = info
 			}
 			duration := time.Duration(ttl)*time.Second + c.node.config.ClientExpiredCloseDelay
-			c.addExpireUpdate(duration)
+			c.addExpireUpdate(duration, true)
 			c.mu.Unlock()
 		} else {
 			go func() { _ = c.close(DisconnectExpired) }()
@@ -1558,7 +1566,7 @@ func (c *Client) handleRefresh(req *protocol.RefreshRequest, cmd *protocol.Comma
 					c.info = info
 				}
 				duration := time.Duration(ttl)*time.Second + c.node.config.ClientExpiredCloseDelay
-				c.addExpireUpdate(duration)
+				c.addExpireUpdate(duration, true)
 				c.mu.Unlock()
 			} else {
 				c.writeDisconnectOrErrorFlush("", protocol.FrameTypeRefresh, cmd, ErrorExpired, started, rw)

